src/rpc/invoice.rs

```diff
@@ -64,12 +64,26 @@ trait InvoiceRpc {
 
 pub(crate) struct InvoiceRpcServerImpl<S> {
     store: S,
-    public_key: Option<PublicKey>,
+    keypair: Option<(PublicKey, SecretKey)>,
 }
 
 impl<S> InvoiceRpcServerImpl<S> {
-    pub(crate) fn new(store: S, public_key: Option<PublicKey>) -> Self {
-        Self { store, public_key }
+    pub(crate) fn new(store: S, config: Option<FiberConfig>) -> Self {
+        let keypair = config.map(|config| {
+            let kp = config
+                .read_or_generate_secret_key()
```

Comment:
Is it safe to be called in multiple threads concurrently?

Reply:
I'm afraid not. Both threads may pass the test that the key does not generate and one thread will use the keypair which is later been overwritten.

https://github.com/chenyukang/fiber/blob/8015b035af68234d74fbcdb8aec0737e2b886693/src/fiber/key.rs#L26

Reply:
I think it's not a real issue, `read_or_generate_secret_key` must be called firstly when starting the network actor in the main.

but yeah, we'd better make it thread-safe with lazy static or something.